wd_core.au3

```diff
@@ -985,56 +985,63 @@ EndFunc   ;==>_WD_GetSource
 ; Parameters ....: $sSession - Session ID from _WD_CreateSession
 ;                  $sCommand - One of the following actions:
 ;                  |
-;                  |ADD    - Create a new cookie. $sOption has to be a JSON string
-;                  |DELETE - Delete a single cookie. The name of the cookie to delete is specified in $sOption
-;                  |GET    - Retrieve the value of a single cookie. The name of the cookie to retrieve has to be specified in $sOption
-;                  |GETALL - Retrieve the values of all cookies
+;                  |ADD       - Create a new cookie. $sOption has to be a JSON string
+;                  |DELETE    - Delete a single cookie. The name of the cookie to delete is specified in $sOption
+;                  |DELETEALL - Delete all cookies
+;                  |GET       - Retrieve the value of a single cookie. The name of the cookie to retrieve has to be specified in $sOption
+;                  |GETALL    - Retrieve the values of all cookies
 ;                  $sOption  - [optional] a string value. Default is ""
 ; Return values .: Success - Requested data returned by web driver.
 ;                  Failure - "" (empty string) and sets @error to one of the following values:
 ;                  - $_WD_ERROR_Exception
 ;                  - $_WD_ERROR_InvalidDataType
+;                  - $_WD_ERROR_InvalidArgue
 ; Author ........: Danp2
-; Modified ......:
-; Remarks .......: Please have a look at WD_Demo.au3 > DemoCookies function for how to add a new cookie
+; Modified ......: mLipok
+; Remarks .......: Please have a look at wd_demo.au3 > DemoCookies function for how to add a new cookie
 ; Related .......:
 ; Link ..........: https://www.w3.org/TR/webdriver#cookies
 ; Example .......: No
 ; ===============================================================================================================================
 Func _WD_Cookies($sSession, $sCommand, $sOption = Default)
 	Local Const $sFuncName = "_WD_Cookies"
 	Local $sResult, $sResponse, $iErr
-
 	If $sOption = Default Then $sOption = ''
 
-	Switch $sCommand
-		Case 'add'
-			$sResponse = __WD_Post($_WD_BASE_URL & ":" & $_WD_PORT & "/session/" & $sSession & "/cookie", $sOption)
-			$iErr = @error
+	$_WD_ERROR_InvalidArgue
+	If $sCommand = 'delete' And (IsString($sOption) = 0 Or StringLen($sOption) = 0 Then $iErr = $_WD_ERROR_InvalidArgue
+	If $sCommand = 'deleteall' And $sOption <> '' Then $iErr = $_WD_ERROR_InvalidArgue

```

Comment:
The first line will break the backward compatibility that I mentioned earlier for `delete` supporting removal of all cookies when $sOption is blank.

Also, can't these be moved within the Case?

Reply:
Fixed: [$_WD_ERROR_InvalidArgue fixed](https://github.com/Danp2/au3WebDriver/pull/195/commits/0651d63a117874ed83c3a4e574d95f308a4d964e)

Reply:
also: [$_WD_ERROR_InvalidArgue moved](https://github.com/Danp2/au3WebDriver/pull/195/commits/338627c02eb8981fd53fe940b09bdb83938ab3c2)

Reply:
> If $sCommand = 'delete' And IsString($sOption) = 0 Then $iErr = $_WD_ERROR_InvalidArgue

This still causes the backward compatibility issue.

Reply:
> If $sCommand = 'delete' And IsString($sOption) = 0 Then $iErr = $_WD_ERROR_InvalidArgue

Why ?
few lines above there is:
`$sOption  - [optional] a string value. Default is ""` 

So as I understand it must be string.

Reply:
See our prior discussion -- https://github.com/Danp2/au3WebDriver/pull/195#issuecomment-1038086089

AFAICS, your change has prevented the prior change made in https://github.com/Danp2/au3WebDriver/pull/195/commits/9d1e6fe6acc10d18d30af8a0d582c4cd33874e40 from working as we discussed.

Reply:
This one ?
```AutoIt
If $sCommand = 'delete' And IsString($sOption) = 0 Then $iErr = $_WD_ERROR_InvalidArgue
.....
If $iErr = $_WD_ERROR_Success Then
	$sResponse = __WD_Delete($sURLSession & "cookie" & ($sOption <> '') ? "/" & $sOption : "")
```

Reply:
Ok. I was either thinking of an earlier instance or I just misinterpreted the statement. Sorry for the confusion.